File: LeetCode/pythonSols/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hashMap = collections.defaultdict()

        for idx,num in enumerate(nums):
            if target-num in hashMap: return [hashMap.get(target-num),idx]
            hashMap[num] = idx

        return [0,0]

File: LeetCode/cppSols/hashmap/twoSum.cc
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;
class Solution{
    public : 
        vector<int> twoSum(vector<int> &nums,int target)
        {
            int i=0;
            unordered_map<int,int> map;
            for(;i<nums.size();i++)
            {
                if(map.count(target-nums[i]))
                    return vector<int>{map[target-nums[i]],i};
                map[nums[i]]=i;
            }
            return {0,0};
        }
};